### PR TITLE
Modified Example.fromJSON() to make JSON processing more powerful.

### PR DESCRIPTION
**Pull request by JahoJiang**
**sample.json:**
```json
{"foods": {"fruits": ["Apple", "Banana"], "vegetables": [{"name": "lettuce"}, {"name": "marrow"}] }}
````

**Before:**
```python3
In [1]: from torchtext import data

In [2]: fields = {'foods.vegetables.name': ('vegs', data.Field())}

In [3]: vegs = data.TabularDataset(path='sample.json', format='json', fields=fields)

ValueError: Specified key foods.vegetables.name was not found in the input data
```

**Now:**
```python3
In [1]: from torchtext import data

In [2]: fields = {'foods.vegetables.name': ('vegs', data.Field())}

In [3]: vegs = data.TabularDataset(path='sample.json', format='json', fields=fields)

In [4]: vegs.examples[0].vegs
Out[4]: ['lettuce', 'marrow']
```

### CODE REVIEW

**Comment by JahoJiang** [comment]
**This failing check may due to the wrong test program.**
**Your data:**
image
**Your test:**
image

That's why the error happened:
image

**Comment by zhangguanheng66** [comment]
@JahoJiang Thanks for the contribution. Do you mind if you could fix the unit test as well?

File: torchtext/data/example.py [hunk 1]
```diff
@@ -11,7 +13,29 @@ class Example(object):
 
     @classmethod
     def fromJSON(cls, data, fields):
-        return cls.fromdict(json.loads(data), fields)
+        # Modified by JahoJiang in 2019.07.17
```
Comment (zhangguanheng66) [inline torchtext/data/example.py]: We don't leave personal information here. People see your contribution from the commit history :).

Comment (JahoJiang) [reply torchtext/data/example.py]: Oops, I forgot to delete it when I submitted it, sorry. 

I will delete the information and fix unit test.

**Comment by JahoJiang** [comment]
> @JahoJiang Thanks for the contribution. Do you mind if you could fix the unit test as well?

I have not seen the unit test code carefully before. When I tried to fix it, I realized that the unit test was correct, It is designed to propose and check KeyError. 
So I refined my code and all checks have passed.

**Pull request closed without merging**